packages/jest-cli/src/run_jest.js

```diff
@@ -96,6 +97,21 @@ export default async function runJest({
 }) {
   const sequencer = new TestSequencer();
   let allTests = [];
+
+  if (changedFilesPromise) {
+    const {repos} = await changedFilesPromise;
+    const noSCM = Object.keys(repos).every(scm => repos[scm].size === 0);
+    if (globalConfig.watch && noSCM) {
```

Comment:
can you move the `globalConfig.watch` check to the outer `if`? No need to do that work if we're not in wath mode

Reply:
done 👍